src/sqlfluff/core/dialects/dialect_bigquery.py

```diff
@@ -59,11 +60,22 @@
 # Unreserved Keywords
 bigquery_dialect.sets("unreserved_keywords").add("SYSTEM_TIME")
 bigquery_dialect.sets("unreserved_keywords").remove("FOR")
+bigquery_dialect.sets("unreserved_keywords").add("STRUCT")
 # Reserved Keywords
 bigquery_dialect.sets("reserved_keywords").add("FOR")
 
 
+# Bracket pairs (a set of tuples)
+bigquery_dialect.sets("bracket_pairs").update(
+    [
+        # NB: Angle brackets can be mistaken, so False
+        ("angle", "LessThanSegment", "GreaterThanSegment", False)
+    ]
+)
+
+
 # BigQuery allows functions in INTERVAL
+@bigquery_dialect.segment(replace=True)
```

Comment:
I'm surprised this line wasn't already present. If existing tests weren't breaking, perhaps new tests should be added?

Reply:
My guess is that there is no test for INTERVAL that uses a function, therefore the ANSI implementation that was being picked up in BigQuery was sufficient